src/simple_edg.rs

```diff
@@ -0,0 +1,158 @@
+/// Defines a hardcoded EDG.
+/// This allows us to easily create EDGs for testing and other purposes. It works well
+/// together with the `edg_assert` macro.
+///
+/// # Example A
+/// A simple EDG with 4 vertices A, B, C, D:
+/// ```
+/// simple_edg![
+///     A => -> {B, C} -> {D};
+///     B => ;
+///     C => -> {B} .> D;
+///     D => -> {};
+/// ];
+///
+/// edg_assert!(A, TRUE);
+/// edg_assert!(B, FALSE);
+/// ```
+/// We list the edges of each vertex, separating each vertex using a semicolon.
+/// The syntax `-> {V, ..., W}` defines a hyper-edge with targets `V`,...,`W`, and the syntax
+/// `.> P` defines a negation edge with target `P`. Declare multiple edges by writing them
+/// after each other (as seen for vertex `A` and `C` above), but note that hyper-edges must
+/// precede negation edges. Additionally, all targets must also be declared in the first line
+/// of the macro, and the vertex declarations and the edge declarations are separated with `::`.
+///
+/// # Example B
+/// By default the created EDG consists of a struct named `SimpleEDG` and enum named
+/// `SimpleVertex`. This can be changed by adding `[EDG_NAME, VERTEX_NAME]` in the start of the
+/// macro's arguments, where `EDG_NAME` is the desired name of the example, and `VERTEX_NAME` is
+/// the desired name of the vertex enum. This allows us to have multiple hardcoded EDGs in the
+/// same scope.
+/// ```
+/// simple_edg![
+///     [MyEDG1, MyVertex1]
+///     A => -> {B} .> {D};
+///     B => -> {D, C};
+///     C => ;
+///     D => -> {C};
+/// ];
+///
+/// edg_assert!([MyEDG1, MyVertex1] A, TRUE);
+/// edg_assert!([MyEDG1, MyVertex1] B, FALSE);
+/// ```
+/// The `edg_assert` macro allows the same naming functionality.
```

Comment:
I do not think we need to mention that here

Reply:
Does not hurt to have.